benches/spec.rs

```diff
@@ -0,0 +1,25 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use pulldown_cmark::{html, Parser};
+use std::fs::{read_dir, read_to_string};
+
+pub fn spec_samples(c: &mut Criterion) {
+    let folder = read_dir("./benches/spec_samples").unwrap();
+    for entry in folder {
+        let entry = entry.unwrap();
+
+        if entry.metadata().unwrap().is_file() {
+            let filename = &entry.file_name().into_string().unwrap();
+            let corpus = read_to_string(entry.path()).unwrap();
+            let mut result = String::with_capacity(corpus.len() * 3 / 2);
+
+            c.bench_function(filename, |b| {
```

Comment:
Could we add a throughput measurement in here? That seems very useful in comparing with other parsers.

Reply:
From my understanding, the throughput benchmarks are usually done with multiple inputs, e.g. how it scales with more input, so I don't think it works here since we'd only have one data point (the size of the input).

I've uploaded an [example report] of the current code though (extract and view `report.html`) -- perhaps it's sufficient?

[example report]: https://drive.google.com/file/d/1h8WpfF8QAs7ih7Ud40kaOa6JzGowyOXU/view?usp=sharing

Reply:
Ok, that makes sense. Let's leave it as is.